stock_putaway_storage_type_strategy/models/stock_location.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, models, fields
+from odoo.osv.expression import AND, OR
+
+
+class StockLocation(models.Model):
+
+    _inherit = 'stock.location'
+
+    pack_storage_strategy = fields.Selection(
+        selection=[('none', 'None'),
+                   ('ordered_locations', 'Ordered Children Locations')],
+        required=True,
+        default='none',
+        string='Packs storage strategy',
+        help='TODO',
+    )
+    display_pack_storage_strategy = fields.Boolean(
+        compute='_compute_display_pack_storage_strategy'
+    )
+
+    @api.depends(
+        'child_ids', 'location_id', 'location_id.pack_storage_strategy'
+    )
```

Comment:
You should check `location_id.display_pack_storage_strategy` as with more than one level, the current depends would not detect the change.

Actually, I'm not sure we should hide the field on locations without children: when creating locations, you'll start by creating the zone/area, configure the strategy, then create the children locations. At worse, setting the strategy on a leaf will have no effect.

Reply:
> setting the strategy on a leaf will have no effect

This is actually why I wanted to hide the field as I don't think it make sense to display something meaningless... :man_shrugging:

Reply:
It has meaning when you are creating the parent before the children (at this point, the parent is a leaf).